Release notes 0.8.3 (#778)
harness/developer-hub: release-notes/chaos-engineering.md

Additionally, the release notes below are only for NextGen SaaS. FirstGen SaaS release notes are available [here](/docs/first-gen/firstgen-release-notes/harness-saa-s-release-notes) and Self-Managed Enterprise Edition release notes are available [here](/release-notes/self-managed-enterprise-edition).
:::

## February 22, 2023, version 0.8.3

### What’s new

* Polling model to communicate between chaos infrastructure and the control plane (CHAOS-644)
    * This release updates the method of communication from web socket to the polling model. This allows the chaos infrastructure to handle higher loads with better scalability.

:::note
From this release onward, chaos infrastructures will communicate with the control plane through the polling model. To allow the existing chaos infrastructure to communicate with the control plane, reconnect or upgrade the chaos infrastructure by redownloading the manifest file.
::: 

* Log service integration with experiment logs (CHAOS-642)

>     * This release adds logs integration into log-service. The logs generated by the chaos experiments persist in GCS even after the experiment pods are deleted from the cluster. It is important to note that these are logs associated with the fault only, and does not include logs of installation and clean-up steps.

Reviewer: ```suggestion
    * This release adds logs integration into log-service. The logs generated by the chaos experiments persist in GCS even after the experiment pods are deleted from the cluster. It is important to note that only logs associated with the fault are retained. Logs for installations and clean-up steps are not retained. 
```

* View and download the report for the runs of the chaos experiment (CHAOS-606)
    * This release allows you to view and download the report (as a PDF file) for all the runs of a chaos experiment. This helps you analyse and store the execution details of the chaos experiment.

* Chaos execution screen shows fault configuration details (CHAOS-1058)

>     * This release displays the fault configuration details along with the probes and logs (previously displayed) on the **View execution details** page. The fault configuration details include target application, and fault tunables that you selected while constructing the experiment.

Reviewer: ```suggestion
    * This release displays the fault configuration details along with the probes and logs (previously displayed) on the **View execution details** page. The fault configuration details include the target application and fault tunables that you selected while constructing the experiment.
```

* Fallback view when tunables are unavailable (CHAOS-1063)
    * This release adds a fallback view when no fault tunables are available when you are constructing a chaos experiment. This fallback view displays the message "No tunables for the selected fault: fault_name.".

> * Chaos configuration step in **Pipelines** shows name of the chaos experiment (CHAOS-986)

Reviewer: ```suggestion
* Chaos configuration step in **Pipelines** shows the name of the chaos experiment (CHAOS-986)
```

* Search functionality when selecting experiments from ChaosHub (CHAOS-1050)
    * This release adds search functionality when selecting an experiment template from ChaosHub. You can also filter the experiments you wish to view or select from the ChaosHub. This allows you to select and run your experiment without having to search through multiple experiments.

* Chaos infrastructure manifest file extension changed to .yaml (CHAOS-1037)
    * This release changes the downloadable chaos infrastructure manifest file extension from .yml to .yaml.

* **Set fault weights** tab moved inside **Tune fault** tab (CHAOS-1077)
    * This release moves the **Set fault weights** tab, which was previously a separate tab, into the **Tune fault** tab. This allows you to tune the fault parameters and set fault weights in a single step rather than navigating through multiple tabs.

* Support for the GitLab connector (CHAOS-35)
    * This release introduces a new connector, namely the GitLab connector, when you wish to connect to a ChaosHub.

### Early access features

* This release does not include any early access features.

### Fixed issues

> * When two faults were being executed in parallel, the timestamp in the **View detailed execution** showed only for the first fault. The second fault showed an empty timestamp field. Now, it has been fixed. (CHAOS-1064)

Reviewer: ```suggestion
* When two faults were being executed in parallel, the timestamp in the **View detailed execution** showed only for the first fault. The second fault showed an empty timestamp field. This issue has been fixed. (CHAOS-1064)
```

 
* When the latest run of a chaos experiment was stopped or had not started yet, the latest run showed the message "This experiment has not been run yet". Now, it has been fixed so that the summary of a chaos experiment shows the latest runs that were successful. (CHAOS-1076)
 

> * When the details of a chaos experiment were being filled, clicking **Cancel** would show a message "Are you sure you want to proceed?" irrespective of whether or not the fields were populated. Now, it has been fixed. (CHAOS-1072)

Reviewer: ```suggestion
* When the details of a chaos experiment were being filled, clicking **Cancel** would show a message "Are you sure you want to proceed?" irrespective of whether or not the fields were populated. This issue has been fixed. (CHAOS-1072)
```

> * When you tried to enable chaos on an infrastructure, clicking anywhere outside the prompt would close the chaos infrastructure selection prompt. Now, it has been fixed. Only by clicking the **X** button at the top does the chaos infrastructure prompt close. (CHAOS-1070)

Reviewer: ```suggestion
* When you tried to enable chaos on an infrastructure, clicking anywhere outside the prompt would close the chaos infrastructure selection prompt. This issue has been fixed. Only by clicking the **X** button at the top does the chaos infrastructure prompt close. (CHAOS-1070)
```

 
* In **Pipeline**, in the **Resilience** tab, the text in the ‘View in chaos module’ button overflowed when the name of the probe was too long. Now, it has been fixed so that the probe name is displayed when you hover on it. (CHAOS-1044)
 
* In **Pipeline**, when you tried to select an experiment in the chaos experiments page, the pagination section overflowed. Now, it has been fixed so that the chaos experiments plage shows two buttons: **Prev** and **Next** to navigate through the pages. (CHAOS-1045)  
 
* In ChaosHubs, the number of experiments in the category tab for the chaos experiments overflowed. Now, this has been fixed. (CHAOS-1053)

> 

Reviewer: Now, it has been fixed => This issue has been fixed.

This is an optional sug.
